src/hooks/usePrevious.ts

```diff
@@ -0,0 +1,11 @@
+import { useEffect, useRef } from "react";
+
+export function usePrevious<T>(value: T): T | undefined {
+  const ref = useRef<T>();
+
+  useEffect(() => {
+    ref.current = value;
+  });
```

Comment:
```suggestion
  }, [value]);
```

Reply:
I'm not sure if it would work this way. Originally this is covered by test.